src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -46,6 +46,9 @@ internal sealed partial class Http2Connection : HttpConnectionBase, IDisposable
         private int _pendingWriters;
 
         private bool _disposed;
+        // anything else than -1 means that connection is being terminated
+        // in which case _lastStreamId is the maximum StreamId which can be processed
+        private int _lastStreamId = -1;
```

Comment:
Nit: can we rename this to _lastValidStreamId?

Reply:
Can we modify the comment so that it specifically refers to GOAWAY, so that it's clear what is meant by "being terminated"? This is only set when we receive GOAWAY, right?

Reply:
re name: the RFC calls the field in GOAWAY "Last-Stream-ID", so it seems reasonable for us to use this in the code. But we should be consistent between this name and the name of the argument in AbortStreams, which is currently "lastValidStreamId"

Reply:
Yeah, that discrepancy is mainly why I commented on it.  I saw "last valid stream id", then saw this "last stream id", and immediately thought they must be referring to different things.

Reply:
Or possibly "_goawayLastStreamId" for clarity that this is from receiving a GOAWAY

Reply:
Did you decide not to rename this?

Reply:
it's already renamed to _lastValidStreamId